integration/tests/overlay-focus-trap.test.js

```diff
@@ -0,0 +1,140 @@
+import { expect } from '@esm-bundle/chai';
+import { fixtureSync, nextRender, oneEvent, tabKeyDown } from '@vaadin/testing-helpers';
+import './not-animated-styles.js';
+import '@vaadin/button/vaadin-button.js';
+import '@vaadin/overlay/vaadin-overlay.js';
+import '@vaadin/radio-group/vaadin-radio-group.js';
+import '@vaadin/text-field/vaadin-text-field.js';
+import { getFocusableElements, isElementFocused } from '@vaadin/component-base/src/focus-utils.js';
+
+describe('focus-trap', () => {
+  let overlay, focusableElements;
+
+  function getFocusedElementIndex() {
+    return focusableElements.findIndex(isElementFocused);
+  }
+
+  describe('elements in light DOM', () => {
+    beforeEach(async () => {
+      overlay = fixtureSync('<vaadin-overlay focus-trap></vaadin-overlay>');
+      await nextRender();
+      overlay.renderer = (root) => {
+        if (!root.firstChild) {
+          root.innerHTML = `
+            <vaadin-text-field></vaadin-text-field>
+            <vaadin-radio-group>
+              <vaadin-radio-button id="radioButton1" label="Button 1"></vaadin-radio-button>
+              <vaadin-radio-button id="radioButton2" label="Button 2"></vaadin-radio-button>
+              <vaadin-radio-button id="radioButton3" label="Button 3"></vaadin-radio-button>
+            </vaadin-radio-group>
+            <vaadin-button>tabindex 0</vaadin-button>
+          `;
+        }
+      };
+      overlay.requestContentUpdate();
+      focusableElements = getFocusableElements(overlay.$.overlay);
+    });
+
+    afterEach(() => {
+      overlay.opened = false;
+    });
+
+    it('should properly detect focusable elements inside the content', () => {
+      expect(focusableElements.length).to.equal(6);
+      expect(focusableElements[0]).to.eql(overlay.$.overlay);
```

Comment:
I guess we can use strict equality here.

```suggestion
      expect(focusableElements[0]).to.equal(overlay.$.overlay);
```